libc/src/__support/OSUtil/uefi/error.h

```diff
@@ -0,0 +1,98 @@
+//===----------- UEFI implementation of error utils --------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIBC_SRC___SUPPORT_OSUTIL_UEFI_ERROR_H
+#define LLVM_LIBC_SRC___SUPPORT_OSUTIL_UEFI_ERROR_H
+
+#include "include/llvm-libc-types/EFI_STATUS.h"
+#include "src/__support/CPP/array.h"
+#include "src/__support/macros/config.h"
+#include <errno.h>
+#include <limits.h>
```

Comment:
We want to avoid including public libc headers directly. Also the internal limits is templated so you can just use `cpp::numeric_limits<EFI_STATUS>::max()` to get the maximum.

```suggestion
#include "hdr/errno_macros.h"
#include "src/__support/CPP/limits.h"
```

Reply:
Using `cpp::numeric_limits` causes me to run into this issue:
```
error: constexpr variable 'UEFI_STATUS_ERRNO_MAP' must be initialized by a constant expression
```

Reply:
I guess `EFI_ERROR_MAX_BIT` as `cpp::numeric_limits<EFI_STATUS>::max()` is what I should be doing

Reply:
Ok, this has been done.